pilot/pkg/security/authz/builder/extauthz.go

```diff
@@ -288,11 +315,27 @@ func generateHeaders(headers []string) *envoy_type_matcher_v3.ListStringMatcher
 	}
 	var patterns []*envoy_type_matcher_v3.StringMatcher
 	for _, header := range headers {
-		patterns = append(patterns, &envoy_type_matcher_v3.StringMatcher{
-			MatchPattern: &envoy_type_matcher_v3.StringMatcher_Exact{
-				Exact: header,
-			},
-		})
+		var pattern *envoy_type_matcher_v3.StringMatcher
+		if strings.HasPrefix(header, "*") {
+			pattern = &envoy_type_matcher_v3.StringMatcher{
+				MatchPattern: &envoy_type_matcher_v3.StringMatcher_Suffix{
+					Suffix: strings.TrimPrefix(header, "*"),
+				},
+			}
+		} else if strings.HasSuffix(header, "*") {
+			pattern = &envoy_type_matcher_v3.StringMatcher{
+				MatchPattern: &envoy_type_matcher_v3.StringMatcher_Prefix{
+					Prefix: strings.TrimSuffix(header, "*"),
+				},
+			}
+		} else {
```

Comment:
I think we can just allow it so that it's consistent with the authz policy. Practically I think this should be fine as we don't hear any user complaints on this.